keras_nlp/models/gemma/gemma_attention.py

```diff
@@ -87,27 +88,22 @@ def build(self, inputs_shape):
             (None, None, self.num_query_heads, self.head_dim)
         )
         self.softmax = keras.layers.Softmax(dtype="float32")
+
+        self.rope_layer = RotaryEmbedding(
+            max_wavelength=10000.0, dtype=self.dtype_policy
+        )
+
         self.built = True
 
-    def _apply_rope(self, x, positions):
+    def _apply_rope(self, x, start_index):
         """Rope rotate q or k."""
-        # TODO: refactor to use RotaryEmbedding layer?
-        max_wavelength = 10000
-        x_shape = ops.shape(x)
-        freq_exponents = (2.0 / x_shape[-1]) * ops.cast(
-            ops.arange(x_shape[-1] // 2, dtype="float32"), self.compute_dtype
+        start_index = ops.cast(start_index, dtype=self.compute_dtype)
```

Comment:
```suggestion
        start_index = ops.cast(start_index, dtype="float32")
```

to address @danielhanchen's comment: https://github.com/keras-team/keras-nlp/pull/1497#issuecomment-1986809020